apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,165 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {ethers} from 'ethers';
+import useSWR from 'swr';
+import {useActionFlow} from '@vaults/contexts/useActionFlow';
+import {Solvers, useSolver} from '@vaults/contexts/useSolver';
+import {getEthZapperContract} from '@vaults/utils';
+import {Button} from '@yearn-finance/web-lib/components/Button';
+import {useSettings} from '@yearn-finance/web-lib/contexts/useSettings';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {useChainID} from '@yearn-finance/web-lib/hooks/useChainID';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {ETH_TOKEN_ADDRESS} from '@yearn-finance/web-lib/utils/constants';
+import {defaultTxStatus} from '@yearn-finance/web-lib/utils/web3/transaction';
+import {useWallet} from '@common/contexts/useWallet';
+import {useAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import {DefaultTNormalizedBN} from '@common/utils';
+
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import type {TNormalizedBN} from '@common/types/types';
+
+function	VaultDetailsQuickActionsButtons(): ReactElement {
+	const {refresh} = useWallet();
+	const {isActive} = useWeb3();
+	const {safeChainID} = useChainID();
+	const {networks} = useSettings();
+	const [txStatusApprove, set_txStatusApprove] = useState(defaultTxStatus);
+	const [txStatusDeposit, set_txStatusDeposit] = useState(defaultTxStatus);
+	const [allowanceFrom, set_allowanceFrom] = useState<TNormalizedBN>();
+	const {
+		selectedOptionFrom, selectedOptionTo,
+		amount, onChangeAmount,
+		maxDepositPossible, isDepositing
+	} = useActionFlow();
+	const {onApprove, onExecuteDeposit, onExecuteWithdraw, currentSolver} = useSolver();
+	const retrieveAllowance = useAllowanceFetcher();
+
+	const shouldUseVanillaAllowance = [Solvers.VANILLA, Solvers.CHAIN_COIN, Solvers.PARTNER_CONTRACT].includes(currentSolver);
+	const isInputTokenEth = selectedOptionFrom?.value === ETH_TOKEN_ADDRESS;
+	const isOutputTokenEth = selectedOptionTo?.value === ETH_TOKEN_ADDRESS;
+	const canInteract = isActive && amount.raw.gt(0) && selectedOptionFrom && selectedOptionTo;
+
+	const spender = useMemo((): TAddress => {
+		if (currentSolver === Solvers.CHAIN_COIN && isOutputTokenEth) {
+			return (toAddress(getEthZapperContract(safeChainID)));
+		} else if (currentSolver === Solvers.PARTNER_CONTRACT) {
```

Comment:
nit: When there's a return inside the condition, I think readably is improved without the `else if`

```suggestion
		}
		
		if (currentSolver === Solvers.PARTNER_CONTRACT) {
```